Framework/Kernel/src/InstrumentInfo.cpp

```diff
@@ -121,16 +121,62 @@ std::string InstrumentInfo::filePrefix(unsigned int runNumber) const {
 }
 
 /// Returns the name of the live listener
-const std::string &InstrumentInfo::liveListener() const {
-  return m_liveListener;
+std::string InstrumentInfo::liveListener(const std::string &name) const {
+  if (!hasLiveListenerInfo())
+    return "";
```

Comment:
This was the previous behaviour (the m_liveListener and m_liveDataAddress members would be empty in this scenario). There is a test in InstrumentInfoTest that checks for it, and I remember several places in the code relied on it.

I think now all those places have been changed to use `hasLiveListenerInfo` instead and so only the test remains.

Do you think we should change it or leave as is?